pull_request.md

```diff
@@ -0,0 +1,99 @@
+- Code review is simple and manageable
+- A clean tree of changes in the commit history
+- Consistency across the commits
+
+We also want to respect the time and effort of those reviewing the work. Keeping PR's small, focused and consistent makes their life easier.
+
+## Always on a branch
+
+No matter how small the change, all changes should be done on a branch and never directly to `master`. This is to support the principle of [Master is always shippable](https://github.com/DEFRA/dst-guides/blob/master/principles/master_is_always_shippable.md)
+
+Clone the repo then create your new branch. For example `git checkout -b add-ea-admin-area-lookup`.
+
+## Start with a commit
+
+Start your new branch with an empty commit.
+
+```bash
+git commit --allow-empty
+```
+
+The template for the commit is the following
+
+```text
+50 character limited title
+
+Link to originating story/bug/card in relevant system (e.g. Jira or Trello)
+
+Description covering why we're making this change, and briefly what the change is.
+```
+
+For example
+
+```text
+Handle empty params in ValidatePreUpdate method
+
+https://eaflood.atlassian.net/browse/WAS-1096
+
+This change fixes an issue found with `validate_pre_update_organisation_address` in that when passed empty parameters it would cause an `undefined method all? for nil:NilClass` error to be thrown.
+```
+
+The key point is that it should cover **the actual change you are intending to make**, and not just repeat **what the backlog story outlined**. The link to the story is added for those that need that greater context.
+
+### Commit message rules
+
+1. Separate subject from body with a blank line
+1. Limit the subject line to 50 characters
+1. Capitalize the subject line
+1. Do not end the subject line with a period
+1. Use the imperative mood in the subject line (*Add ability* not *Added ability*)
+1. Use the body to explain *what and why* vs. *how*
+
+N.B. The source for these is [How to write a Git commit message](http://chris.beams.io/posts/git-commit/) which you're encouraged to read.
+
+## Let everyone know
+
+Next push your branch.
+
+```bash
+git push -u origin add-ea-admin-area-lookup
+```
+
+We create a PR right from the start so that your proposed change is visible to all. This means everyone can feedback if they see any issues, and give help and advice.
+
+Once pushed you'll need to go to GitHub to actually create the PR. Typically it will highlight the new branch in the UI and provide a quick option to create the PR. You'll then see GitHub automatically takes your first empty commit and uses it to populate the pull request title and description. Nice! Assign yourself to the PR so everyone knows the work is with you.
+
+## Now code
+
+Now get on and code. You should commit frequently and push often. Don't worry too much about your commit messages here. But don't ignore them completely as they will be used as part of the PR process (so keep `WIP`, `Still WIP`, and `More WIP!` to a minimum thank you!)
+
```

Comment:
Is it worth providing a quick word of guidance here to say that you can still do a few "WIP" commits in order to keep your code safely committed but then squash them into a single commit with a more helpful description?

Reply:
I can expand the description to include why you might want to use a `WIP` as a commit message, and that these will be squashed.

I could get into the whole interactive rebase thing, where you can tidy up your commits ahead of the PR. But I think that's a bit advanced for this guide and would just confuse things. But I could do something on it in a later guide "Getting ready for review - advanced!"

Reply:
Yes I agree, it's not the place for a detailed discussion on interactive rebasing, I just wanted to make sure that people weren't put off from committing their code "little and often" if it helped their workflow.

Basically, I liked the general "Don't worry too much about your commit messages here" vibe.